esp/smc/SMCLib/TpWrapper.cpp

```diff
@@ -1824,31 +1824,71 @@ void CTpWrapper::getAttPath(const char* Path,StringBuffer& returnStr)
+
+    if (isEmptyString(serviceType) || strieq(serviceType, "roxie"))
+        getRoxieServices(envRoot, serviceName, services);
+    getESPServices(envRoot, serviceType, serviceName, services);
+}
+
+void CTpWrapper::getRoxieServices(IPropertyTree* environmentRoot, const char* serviceName, IArrayOf<IConstHPCCService>& services)
+{
+    if (!isEmptyString(serviceName))
     {
-        IPropertyTree& service = itr->query();
-        //Only show the public services for now
-        if (!service.getPropBool("@public"))
-            continue;
+        VStringBuffer xpath("Software/RoxieCluster[@name=\"%s\"]", serviceName);
+        getRoxieService(environmentRoot->queryPropTree(xpath.str()), services);
+        return;
+    }
+    Owned<IPropertyTreeIterator> clusters = environmentRoot->getElements("Software/RoxieCluster");
+    ForEach(*clusters)
+    {
+        IPropertyTree& cluster = clusters->query();
+        getRoxieService(&cluster, services);
+    }
+}
 
-        const char* type = service.queryProp("@type");
-        if (isEmptyString(type) || (!isEmptyString(serviceType) && !strieq(serviceType, type)))
+void CTpWrapper::getRoxieService(IPropertyTree* clusterTree, IArrayOf<IConstHPCCService>& services)
+{
+    Owned<IEspHPCCService> svc = createHPCCService();
+    svc->setName(clusterTree->queryProp("@name"));
+    svc->setType("roxie");
+
+    Owned<IPropertyTreeIterator> roxieFarms = clusterTree->getElements("RoxieFarmProcess");
+    ForEach(*roxieFarms)
+    {
+        IPropertyTree& roxieFarm = roxieFarms->query();
+        unsigned port = roxieFarm.getPropInt("@port", ROXIE_SERVER_PORT);
+        if (port == 0)
             continue;
+        svc->setPort(port);
+        break;
+    }
+    services.append(*svc.getLink());
+}
 
-        const char* name = service.queryProp("@name");
-        if (isEmptyString(name) || (!isEmptyString(serviceName) && !strieq(serviceName, name)))
+void CTpWrapper::getESPServices(IPropertyTree* environmentRoot, const char* serviceType, const char* serviceName, IArrayOf<IConstHPCCService>& services)
+{
+    Owned<IPropertyTreeIterator> espBindings = environmentRoot->getElements("Software/EspProcess/EspBinding");
+    ForEach(*espBindings)
+    {
+        IPropertyTree& espBinding = espBindings->query();
+        const char* type = espBinding.queryProp("@service");
+        if (!isEmptyString(serviceType) && !strieq(serviceType, type))
+            continue;
+        const char* name = espBinding.queryProp("@name");
+        if (!isEmptyString(serviceName) && !strieq(serviceName, name))
+            continue;
+        unsigned port = espBinding.getPropInt("@port");
+        if (port == 0)
             continue;
 
         Owned<IEspHPCCService> svc = createHPCCService();
         svc->setName(name);
         svc->setType(type);
-        svc->setPort(service.getPropInt("@port"));
-        if (service.getPropBool("@tls"))
-            svc->setTLSSecure(true);
+        svc->setPort(port);
```

Comment:
Is the loss of the setTLSSecure() deliberate?

Reply:
This is for bare metal environment only. Is the tls supported in a bare metal environment? (I haven't found it in ConfigMgr).

Reply:
For Roxie service, I guess the TLS Secure should be true if the protocol = "ssl".